### PR TITLE
fix: Correct search logic to include `sort()` which will return the correct result

### PR DESCRIPTION
**Pull request by jotasixto**
## Description
Getting the test workflow to run successfully.

## Motivation and Context
Reviewing the test workflow we can see that the validation is always executed in the root directory. Therefore we always get the message `Terraform initialized in an empty directory!`, [link of a case](https://github.com/clowdhaus/terraform-min-max/actions/runs/3453154522/jobs/5763535808).
For this I used the option to modify the working directory of the workflow steps.

Furthermore, to be able to carry out the tests, I had to add the ordering logic by size again. I'm sorry, but I'm not clear on the reason for the exclusions of the regex filter. This github action is executed at the root path of the checkout, therefore there will be no "versions.tf" file, but yes the "tests/versions.tf" or similar, always with the subpath included in the value. 
For example in ./tests the github action gets this array `["tests/versions.tf","tests/terraform-aws-alb/wrappers/versions.tf","tests/terraform-aws-alb/versions.tf","tests/terraform-aws-alb/examples/complete-nlb/versions.tf","tests/terraform-aws-alb/examples/complete-alb/versions.tf","tests/0.13/versions.tf","tests/0.12/versions.tf"]` in the versionConstraintSearch function.

## How Has This Been Tested?
I have been testing [this fork](https://github.com/jotasixto/terraform-min-max/actions) to see the behavior

### CODE REVIEW

**Comment by jotasixto** [comment]
I think the error about the path ./tests/terraform-aws-alb/wrappers testing is correct, the main.tf has a reference to the path ../ in the source property. This causes the terraform version 0.13.1 to be established in wrappers, but in turn the source references have 1.0.0, causing the conflict. This case will also occur in [this PR](https://github.com/terraform-aws-modules/terraform-aws-alb/pull/264).

**Comment by jotasixto** [comment]
On the other hand [this error](https://github.com/clowdhaus/terraform-min-max/actions/runs/3455215887/jobs/5767080992) is also an correct case, aws provider version 4.27 requires at least terraform version 0.13 or later.

**Comment by bryantbiggs** [comment]
> I think [the error](https://github.com/clowdhaus/terraform-min-max/actions/runs/3455215887/jobs/5767080871) about the path ./tests/terraform-aws-alb/wrappers testing is correct, the main.tf has a reference to the path ../ in the source property. This causes pthe terraform version 0.13.1 to be established in wrappers, but in turn the source references have 1.0.0, causing the conflict. This case will also occur in https://github.com/terraform-aws-modules/terraform-aws-alb/pull/264.

Correct - this is the source of the issue we are trying to mitigate. I don't want to evaluate the `wrappers` directory at all when validating with min version check. The only thing that will end up evaluating that directory will be the ["max" workflow](https://github.com/terraform-aws-modules/terraform-aws-alb/blob/858ae988b3efe72f7d69335ca065c09cbb4c76d7/.github/workflows/pre-commit.yml#L59) because the wrapper pre-commit hook will always be set to [this value](https://github.com/antonbabenko/pre-commit-terraform/blob/0e07e28b129a077467306452eeba40965c13f303/hooks/terraform_wrapper_module_for_each.sh#L41) currently

so to share a bit more about our workflows to get a better view:
1. We want to find all of the "root" directories where a set of Terraform configuration files live. This could be a module root directory, an example root directory, or a project root directory. We find ALL of these root directories by searching through `*.tf` files to find the `required_version` (since you can only have one of these, but you may have none in which our pre-commit hook will end up complaining if you do not have it - so all roads lead back to this as signfiying a root directory). We want to find these various root directories to a) be able to parallelize checks and b) we want to evaluate them on their own (note - this does not do full validation of all dependency version requirements currently)
2. Once we have our directories, we then have two separate workflows. One set where we look at each directory and determine the [minimum supported terraform version](https://github.com/terraform-aws-modules/terraform-aws-alb/blob/858ae988b3efe72f7d69335ca065c09cbb4c76d7/.github/workflows/pre-commit.yml#L26), and one set where we use the maximum supported version and evaluate the entire project](https://github.com/terraform-aws-modules/terraform-aws-alb/blob/858ae988b3efe72f7d69335ca065c09cbb4c76d7/.github/workflows/pre-commit.yml#L59). 

It is in this 2nd step where we evaluate the minimum version where we want to ensure we are not evaluating anything under the `wrappers` sub-directory. You can also see the relationship between this action and https://github.com/clowdhaus/terraform-composite-actions/tree/main/directories - we have to ensure the root directory search logic is the same or else we will have a mis-match in evaluation 

So now I think we are close, I believe both of the actions search logic is nearly identical, now its just getting this action to return the versions as intended

File: src/search.ts [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+// We only want files at the current directory level, exlcude sub-directories
+const regExprSubDirectory = /\//;
 
 export async function versionConstraintSearch(dir: string): Promise<string> {
   const files = await findInFiles.find('required_versions*s*', dir, '.tf$');
-  const key = Object.keys(files).filter(word => !regExprSubDirectory.test(word))[0];
+
+  const key = Object.keys(files)
+    .sort((a, b) => a.length - b.length)
+    .filter(word => !regExprSubDirectory.test(word))[0];
```
Comment (jotasixto) [inline src/search.ts:12]: I can see you're still fighting this :sweat: 
By denying the regex you are removing any case that is not from the root path of the checkout, that is from the project root and not from the sub path. We must take into account that findInFiles returns the path from the root like `tests/0.13/versions.tf`, working with array like this: `["tests/0.13/versions.tf","tests/0.12/versions.tf","tests/terraform-aws-alb/versions.tf","tests/terraform-aws-alb/wrappers/versions.tf","tests/terraform-aws-alb/examples/complete-nlb/versions.tf","tests/terraform-aws-alb/examples/complete-alb/versions.tf"]`.
With this negation we do not get any file.

I have [this test adding](https://github.com/jotasixto/terraform-aws-alb/pull/1) the skip of the wrappers dir in the target workflow, but the truth is not very elegant.

Another option is to add working directory logic to this piece of code, to run from the root of the desired final directory.

Comment (bryantbiggs) [reply src/search.ts:12]: ah yes, I can't ignore all sub-directories, just paths with `wrappers` in it. thanks

Comment (bryantbiggs) [reply src/search.ts]: I don't even know if I need the regexp filtering, I think I see where you're coming from with the length check. But everything seems to be jiving now and we're doubly ensured we won't evaluate `wrappers/` 😅 - thanks a ton for your help and watching as I struggle through this!!!

**Review by bryantbiggs — APPROVED** [review]

**Pull request merged**